arroyo/backends/kafka/commit.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+from typing import Optional
+
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.commit import Commit
+from arroyo.types import Partition, Topic
+from arroyo.utils.codecs import Codec
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+class CommitCodec(Codec[KafkaPayload, Commit]):
+    def encode(self, value: Commit) -> KafkaPayload:
+        assert value.orig_message_ts is not None
+
+        return KafkaPayload(
+            f"{value.partition.topic.name}:{value.partition.index}:{value.group}".encode(
+                "utf-8"
+            ),
+            f"{value.offset}".encode("utf-8"),
+            [
+                (
+                    "orig_message_ts",
+                    datetime.strftime(value.orig_message_ts, DATETIME_FORMAT).encode(
+                        "utf-8"
+                    ),
+                )
+            ],
+        )
+
+    def decode(self, value: KafkaPayload) -> Commit:
+        key = value.key
+        if not isinstance(key, bytes):
```

Comment:
I believe they are nullable. I should add that this is all being moved into the kafka backend without any code changes (this is to avoid circular dependencies and because this code is only relevant to the kafka specific backend anyway), and would prefer to avoid making functional changes here.